core/src/main/java/org/infinispan/interceptors/impl/IsMarshallableInterceptor.java

```diff
@@ -93,6 +94,15 @@ public Object visitComputeIfAbsentCommand(InvocationContext ctx, ComputeIfAbsent
       return invokeNext(ctx, command);
    }
 
+   @Override
+   public Object visitReadWriteKeyCommand(InvocationContext ctx, ReadWriteKeyCommand command) throws Throwable {
+      if (isUsingAsyncStore(ctx, command)) {
+         checkMarshallable(command.getKey());
+         checkMarshallable(command.getFunction());
```

Comment:
From the point of async store, it is not necessary to have the *function* serializable but the resulting entry. So the check does not belong here.
The same holds for compute commands.